react-app/webpack/webpack.common.js

```diff
@@ -0,0 +1,39 @@
+const path = require('path')
+const loaders = require('./loaders');
+const plugins = require('./plugins');
+
+module.exports = {
+  entry: path.resolve(__dirname, '../src', 'index.js'),
+  output: {
+    path: path.resolve(__dirname, '../dist'),
+    filename: 'bundle.js'
+  },
+  devServer: {
+    contentBase: path.resolve(__dirname, '../dist'),
+    open: true,
+    clientLogLevel: 'silent',
+    port: 9000,
+    historyApiFallback: true,
+    hot: true
+  },
+  module: {
+    rules: [
+      loaders.JSXLoader,
+      loaders.SVGSpriteLoader,
+      loaders.CSSLoader,
+      loaders.SASSLoader,
+      loaders.ImageLoader,
+      loaders.FontLoader,
+      //loaders.ESLint,
+    ],
+  },
+  plugins: [
+    plugins.ImageminPlugin,
+    plugins.SpriteLoaderPlugin,
+    plugins.MiniCssExtractPlugin,
+    plugins.ProgressPlugin,
+    plugins.CleanWebpackPlugin,
+    plugins.HotModuleReplacement,
```

Comment:
I don't think you're supposed to use hmr for `production` builds, just like optimizing your CSS (with MiniCssExtractPlugin) delays unnecessarily re-loads. You'd be much better positioned having those within respective subconfigs. 

Speaking of `MiniCssExtractPlugin`, it is often combined with either [`CssMinimizerWebpackPlugin`](https://webpack.js.org/plugins/css-minimizer-webpack-plugin) or [`OptimizeCssAssetsWebpackPlugin`](https://www.npmjs.com/package/optimize-css-assets-webpack-plugin) to better optimize your production build.